beacon_node/eth2-libp2p/src/behaviour/mod.rs

```diff
@@ -707,11 +824,61 @@ impl<TSpec: EthSpec> Behaviour<TSpec> {
     }
 }
 
+/* Public API types */
+
+/// The type of RPC requests the Behaviour informs it has received and allows for sending.
+// NOTE: This an application-level wrapper over the lower network leve requests that can be sent.
+//       The main difference is the absense of the Ping and Metadata protocols, which don't leave
+//       the Behaviour. For all protocols managed by RPC see `RPCRequest`.
+#[derive(Debug)]
+pub enum Request {
+    /// A Status message.
+    Status(StatusMessage),
+    /// A Goobye message.
+    Goodbye(GoodbyeReason),
+    /// A blocks by range request.
+    BlocksByRange(BlocksByRangeRequest),
+    /// A request blocks root request.
+    BlocksByRoot(BlocksByRootRequest),
+}
+
+/// The type of RPC responses the Behaviour informs it has received, and allows for sending.
+// NOTE: This an application-level wrapper over the lower network level responses that can be sent.
+//       The main difference is the absense of Pong and Metadata, which don't leave the Behaviour.
+//       For all protocol reponses managed by RPC see `RPCResponse` and `RPCCodedResponse`.
+#[derive(Debug)]
+pub enum Response<TSpec: EthSpec> {
+    /// A Status message.
+    Status(StatusMessage),
+    /// A response to a get BLOCKS_BY_RANGE request. A None response signals the end of the batch.
+    BlocksByRange(Option<Box<SignedBeaconBlock<TSpec>>>),
+    /// A response to a get BLOCKS_BY_ROOT request.
+    BlocksByRoot(Option<Box<SignedBeaconBlock<TSpec>>>),
+}
+
 /// The types of events than can be obtained from polling the behaviour.
 #[derive(Debug)]
 pub enum BehaviourEvent<TSpec: EthSpec> {
-    /// A received RPC event and the peer that it was received from.
-    RPC(PeerId, RPCEvent<TSpec>),
+    /// An RPC Request that was sent failed
+    RPCFailed {
+        /// The id of the failed request.
+        id: RequestId,
+        /// The peer_id to which this request was sent.
+        peer_id: PeerId,
+        /// The error that occurred.
+        // TODO: error not needed? remove
+        error: RPCError,
```

Comment:
I'd like to keep it, because I currently log in in the network crate, next to client type that did it. I.e prysm failed because blah. Makes things easier to debug in the context of the RPC's use